frontend/src/metabase/visualizations/click-actions/actions/HideColumnAction/HideColumnAction.tsx

```diff
@@ -1,12 +1,9 @@
 import { t } from "ttag";
 
 import { onUpdateVisualizationSettings } from "metabase/query_builder/actions";
-import {
-  findColumnSettingIndex,
-  getColumnSettingsWithRefs,
-} from "metabase/visualizations/components/settings/ChartSettingTableColumns/utils";
 import type { LegacyDrill } from "metabase/visualizations/types";
 import * as Lib from "metabase-lib";
+import { findColumnSettingIndexesForColumns } from "metabase-lib/queries/utils/dataset";
```

Comment:
1. Create a question [like this](https://github.com/metabase/metabase/assets/6830683/dd28c30b-34af-47bd-9d62-8327b081a953)
2. Visualize it
3. Open visualization settings
4. Click "Add or remove columns"
5. Click to enable "Product" > "Title" column

:x: Column is not marked as enabled

It works in `master`.

----

1. Create a question [like this](https://github.com/metabase/metabase/assets/6830683/dd28c30b-34af-47bd-9d62-8327b081a953)
2. Visualize it
3. Open visualization settings
4. Click "Add or remove columns"
5. Click to enable "Product" > "Title" column
6. Click to enable "Product 2" > "Category" column

[Error: `Cannot find matching FK Table ID for FK Field 37 "Product ID"`](https://github.com/metabase/metabase/assets/6830683/946b7ddb-9265-4722-9ac8-0d4eeac732e7)

This is broken in `master` as well.

Reply:
I think this is what Alex called "broken" on slack, so there expected to be follow-up PRs

at the same time, seems we don't have tests covering such cases

Reply:
The first case works in master accidentally. It's definitely a MLv2 bug